### PR TITLE
fix(framework): fix keras single input edge case

### PR DESCRIPTION
**Pull request by larme**
## What does this PR address?

When handling multi input, we should treat single input separately with multiple input. We used to put all inputs in a list for multi input support, but for some edge case this not works for a single input.

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

Feel free to tag members/contributors who can help review your PR.

### CODE REVIEW

File: bentoml/_internal/frameworks/keras.py [hunk 1]
```diff
@@ -317,7 +317,14 @@ def _run_method(
...
                 params = params.map(_mapping)
-                res: "tf_ext.EagerTensor" | "ext.NpNDArray" = raw_method(params.args)
+                if len(params.args) == 1:
+                    res: "tf_ext.EagerTensor" | "ext.NpNDArray" = raw_method(
+                        *params.args
+                    )
+                else:
+                    res: "tf_ext.EagerTensor" | "ext.NpNDArray" = raw_method(
+                        params.args
+                    )
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/keras.py:327]: Maybe a nicer way is to make the params.args to be a list
```python
if len(params.args) == 1:
    args = [params.args]
else:
    args = params.args

res = raw_method(*args)
```

Comment (larme) [reply bentoml/_internal/frameworks/keras.py]: the intention logic is slightly different but I made a similar change

**Review by aarnphm — APPROVED** [review]
Can you also write a test for this edge case? otherwise LGTM

**Review by ssheng — APPROVED** [review]
